File: evals/workflow_value/results/2026-09-25/cells/01/submitted-project/src/range_py_monitor/test/test_conversion.py
import math

from range_interfaces.msg import RangeReading

from range_py_monitor.conversion import normalize


def test_valid_reading_returns_metres():
    msg = RangeReading()
    msg.distance_m = 2.5
    msg.valid = True
    assert normalize(msg) == 2.5


def test_invalid_reading_returns_nan():
    msg = RangeReading()
    msg.distance_m = 2.5
    msg.valid = False
    assert math.isnan(normalize(msg))

File: evals/workflow_value/results/2026-09-25/cells/01/submitted-project/src/range_cpp_monitor/test/test_conversion.cpp
#include <gtest/gtest.h>

#include <cmath>

#include "range_cpp_monitor/conversion.hpp"

TEST(Conversion, ValidReadingReturnsMetres)
{
  range_interfaces::msg::RangeReading msg;
  msg.distance_m = 2.5;
  msg.valid = true;
  EXPECT_DOUBLE_EQ(range_cpp_monitor::normalize(msg), 2.5);
}

TEST(Conversion, InvalidReadingReturnsNan)
{
  range_interfaces::msg::RangeReading msg;
  msg.distance_m = 2.5;
  msg.valid = false;
  EXPECT_TRUE(std::isnan(range_cpp_monitor::normalize(msg)));
}
